cmd/config-current-notify.go

```diff
@@ -0,0 +1,1188 @@
+	if err := cfg.Validate(); err != nil {
+		return err
+	}
+
+	s[config.NotifyMQTTSubSys][mqttName] = []config.KV{
+		{
+			Key: config.StateKey,
+			Value: func() string {
+				if cfg.Enable {
+					return config.StateEnabled
+				}
+				return config.StateDisabled
+			}(),
+		},
+		{
+			Key:   target.MqttBroker,
+			Value: cfg.Broker.String(),
+		},
+		{
+			Key:   target.MqttTopic,
+			Value: cfg.Topic,
+		},
+		{
+			Key:   target.MqttQoS,
+			Value: fmt.Sprintf("%d", cfg.QoS),
+		},
+		{
+			Key:   target.MqttUsername,
+			Value: cfg.User,
+		},
+		{
+			Key:   target.MqttPassword,
+			Value: cfg.Password,
+		},
+		{
+			Key:   target.MqttReconnectInterval,
+			Value: cfg.MaxReconnectInterval.String(),
+		},
+		{
+			Key:   target.MqttKeepAliveInterval,
+			Value: cfg.KeepAlive.String(),
+		},
+		{
+			Key:   target.MqttQueueDir,
+			Value: cfg.QueueDir,
+		},
+		{
+			Key:   target.MqttQueueLimit,
+			Value: strconv.Itoa(int(cfg.QueueLimit)),
+		},
+	}
+
+	return nil
+}
+
+func (s serverConfig) GetNotifyMQTT() map[string]target.MQTTArgs {
+	mqttTargets := make(map[string]target.MQTTArgs)
+	for k, kvs := range s[config.NotifyMQTTSubSys] {
+		enabled := kvs.Get(config.StateKey) == config.StateEnabled
+		if !enabled {
+			continue
+		}
+
+		brokerURL, err := xnet.ParseURL(kvs.Get(target.MqttBroker))
+		if err != nil {
+			logger.LogIf(context.Background(), err)
+			continue
+		}
+		reconnectInterval, err := time.ParseDuration(kvs.Get(target.MqttReconnectInterval))
+		if err != nil {
+			logger.LogIf(context.Background(), err)
+			continue
+		}
+		keepAliveInterval, err := time.ParseDuration(kvs.Get(target.MqttKeepAliveInterval))
+		if err != nil {
+			logger.LogIf(context.Background(), err)
+			continue
+		}
+		queueLimit, err := strconv.Atoi(kvs.Get(target.MqttQueueLimit))
+		if err != nil {
+			logger.LogIf(context.Background(), err)
+			continue
+		}
+		// Parse uint8 value
+		qos, err := strconv.ParseUint(kvs.Get(target.MqttQoS), 10, 8)
+		if err != nil {
+			logger.LogIf(context.Background(), err)
+			continue
+		}
+
+		mqttArgs := target.MQTTArgs{
+			Enable:               enabled,
+			Broker:               *brokerURL,
+			Topic:                kvs.Get(target.MqttTopic),
+			QoS:                  byte(qos),
+			User:                 kvs.Get(target.MqttUsername),
+			Password:             kvs.Get(target.MqttPassword),
+			MaxReconnectInterval: reconnectInterval,
+			KeepAlive:            keepAliveInterval,
+			RootCAs:              globalRootCAs,
+			QueueDir:             kvs.Get(target.MqttQueueDir),
+			QueueLimit:           uint64(queueLimit),
+		}
+
+		if err = mqttArgs.Validate(); err != nil {
+			logger.LogIf(context.Background(), err)
+			continue
+		}
+		mqttTargets[k] = mqttArgs
+	}
+	return mqttTargets
+}
+
+func (s serverConfig) SetNotifyMySQL(sqlName string, cfg target.MySQLArgs) error {
+	if err := cfg.Validate(); err != nil {
+		return err
+	}
+
+	s[config.NotifyMySQLSubSys][sqlName] = []config.KV{
```

Comment:
Are queueDir & queueLimit are missing here ? I can see them in the original config.json

Reply:
Yes they are missing need to fix them @vadmeste